tests/sqlparser_bigquery.rs

```diff
@@ -87,6 +87,424 @@ fn parse_raw_literal() {
+        expr_from_projection(&select.projection[0])
+    );
+
+    assert_eq!(
+        &Expr::Struct {
+            expr: Box::new(StructExpr(vec![ExprWithFieldName {
+                expr: Expr::TypedString {
+                    data_type: DataType::Timestamp(None, TimezoneInfo::None),
+                    value: "2008-12-25 15:30:00 America/Los_Angeles".to_string()
+                },
+                field_name: None
+            }])),
+            type_ann: Some(vec![StructTypeAnn {
+                field_name: None,
+                field_type: StructFieldType::Timestamp
+            }])
+        },
+        expr_from_projection(&select.projection[1])
+    );
+
+    assert_eq!(
+        &Expr::Struct {
+            expr: Box::new(StructExpr(vec![ExprWithFieldName {
+                expr: Expr::TypedString {
+                    data_type: DataType::Time(None, TimezoneInfo::None),
+                    value: "15:30:00".to_string()
+                },
+                field_name: None
+            }])),
+            type_ann: Some(vec![StructTypeAnn {
+                field_name: None,
+                field_type: StructFieldType::Time
+            }])
+        },
+        expr_from_projection(&select.projection[2])
+    );
+
+    let sql = r#"SELECT STRUCT<NUMERIC>(NUMERIC '1'), STRUCT<BIGNUMERIC>(BIGNUMERIC '1')"#;
+    let select = bigquery().verified_only_select(sql);
+    assert_eq!(2, select.projection.len());
+
+    assert_eq!(
+        &Expr::Struct {
+            expr: Box::new(StructExpr(vec![ExprWithFieldName {
+                expr: Expr::TypedString {
+                    data_type: DataType::Numeric(ExactNumberInfo::None),
+                    value: "1".to_string()
+                },
+                field_name: None
+            }])),
+            type_ann: Some(vec![StructTypeAnn {
+                field_name: None,
+                field_type: StructFieldType::Numeric
+            }])
+        },
+        expr_from_projection(&select.projection[0])
+    );
+
+    assert_eq!(
+        &Expr::Struct {
+            expr: Box::new(StructExpr(vec![ExprWithFieldName {
+                expr: Expr::TypedString {
+                    data_type: DataType::BigNumeric(ExactNumberInfo::None),
+                    value: "1".to_string()
+                },
+                field_name: None
+            }])),
+            type_ann: Some(vec![StructTypeAnn {
+                field_name: None,
+                field_type: StructFieldType::BigNumeric
+            }])
+        },
+        expr_from_projection(&select.projection[1])
+    );
+
+    // TODO: support "ARRAY in STRUCT", "STRUCT in STRUCT"
+    // let sql = r#"SELECT STRUCT<STRUCT<STRING>>(STRUCT("foo"))"#;
+    // let select = bigquery().verified_only_select(sql);
+    // assert_eq!(1, select.projection.len());
+
+    // let sql = r#"SELECT STRUCT<ARRAY<STRING>>(["foo"])"#;
+    // let select = bigquery().verified_only_select(sql);
+    // assert_eq!(1, select.projection.len());
```

Comment:
I want to postpone this. I have no idea to parse this now because Lexer treats ">>" as shiftleft.

Reply:
@togami2864  I think I may have misunderstood your intent -- do you want to postpone the entire PR or just this portion of the syntax?